AttrTable.cc

```diff
@@ -1197,10 +1316,10 @@ void AttrTable::simple_print(ostream &out, string pad, Attr_iter i, bool derefer
 #endif
         vector<string> *sxp = (*i)->attr;
         vector<string>::iterator last = sxp->end() - 1;
-        for (vector<string>::iterator i = sxp->begin(); i != last; ++i) {
-            write_string_attribute_for_das(out, *i, ", ");
+        for (vector<string>::iterator i_s = sxp->begin(); i_s != last; ++i_s) {
```

Comment:
Since this is new code and sxp is a vector<>, you could use the newer form of the for loop:
for (auto &attr: sxp) 
    if (&elem != &item_vector.back())
         write_string_attribute_for_das(out, attr, ", ", (*i)->is_utf8_str);
    else
         write_string_attribute_for_das(out, attr, ";\n", (*i)->is_utf8_str);

The nice thing about these loops is that they are concise and unlikely to have off by one errors

Reply:
I added a section of code according to your suggestion. However, it causes failure of a few tests, which doesn't make sense to me. I leave an #if 0 #endif code block for future investigation.